### DISCUSSION TITLE
v3.0.1

### DISCUSSION THREAD

**Discussion by TooTall18T** [TooTall18T/tool_length_probe#11] [Announcements]
Stable version of [tool_length_probe](https://github.com/TooTall18T/tool_length_probe) subroutine for Probe Basic.

<hr /><em>This discussion was created from the release <a href='https://github.com/TooTall18T/tool_length_probe/releases/tag/3.0.1'>v3.0.1</a>.</em>

**Comment by BSTguitars** [comment]
Any chance you'll update for the current version ? I can do testing .

**Reply by TooTall18T** [reply]
Which version do you mean?
For PB 0.6.0-stable is my version 5.
If you are on the PB develop version 0.6.0-37 or higher, take a look in my develop branch. 
I checked that version up to PB 0.6.0-115.

**Reply by BSTguitars** [reply]
I was confused by "The routines were tested with LinuxCNC 2.9.3 and Probe Basic 0.6.0-018" 

I also had trouble trying to figure out what version was current stable for probe and was not sure if Linux 2.9.4 would be an issue .

**Reply by TooTall18T** [reply]
LCNC 2.9.4 is only a bug fix. 
For my routine the version of LinuxCNC is not very important. You can run it on nearly every version. But Probe Basic need the correct version of LinuxCNC.

**Reply by BSTguitars** [reply]
Ok so the routine dose work with current stable cool.  Sorry for the dumb question ran into some issues and started getting over whelmed.

**Reply by TooTall18T** [reply]
No problem.

**Reply by BSTguitars** [reply]
Me again . lol so every things seems to be working. 

<tool_min_dis> = 10  
This I think is causing some issues . 

I'm tried a bunch of different values and some times it works some times it dose not. gives me a warning about the exceeding z travel or it will move down then stop with out probing depending on the tool offset from tool table. 
Some times it will probe with no issue then if I re run it I'll get an error after it records the value. 

I assume the value I put in that filed should be 5.5 - the longest tool I'll use ?  I included a screen shot of the probe screen showing what I have in there. 

![image](https://github.com/user-attachments/assets/eeb0a05b-e47d-41d3-adf5-54f9e35c1cb1)

My from g53 z0 to probe trigger is g53 z-5.5 

I'll use tools that stick out .25" to 3" .

Since I'm using a spindle with a er collet my tool lengths will vary greatly . 

Sorry for being a pain my Brian is fired after wasting 2 days to figure out my rs485 port on the spindle is fried.

**Reply by TooTall18T** [reply]
<tool_min_dis> is the distance between a known tool and the tool length sensor. 
In my case I have 6mm. When I install the tool at the same height as before, the machine moves the tool that distance above the sensor and starts the measurement. 
If the tool is unknown (length 0) the machine starts at "Spindle zero" and moves all the way down.

"<tool_min_dis> = 10" is a default value and need to be set individually. 10(mm) is a distance that is probably safe and the beginning, but need to be adjusted. 

The values are in your machine units.

If you use the same collet at the same tool, you can use the function "tool table". 
But if you use different lengths collects you can get problems if the variations are bigger than tool_min_dis

**Reply by BSTguitars** [reply]
all right gotcha.  so if im lazy and dont want to roughly fill oout the tool table before hand just set them to zero and i dont need to worry about a crash or error.  ill set it to .4 roughly 10mm in the settings tho. Thank you so much Again.
